frontend/app/src/composables/api/reports/index.ts

```diff
@@ -143,14 +146,17 @@ export function useReportsApi(): UseReportsApi {
   };
 
   const fetchReportEvents = async (
-    reportId: number,
-    page: { limit: number; offset: number },
-  ): Promise<ProfitLossReportEvents> => {
-    const response = await api.instance.post<ActionResult<ProfitLossReportEvents>>(`/reports/${reportId}/data`, page, {
-      validateStatus: validStatus,
-    });
+    payload: ProfitLossEventsPayload,
+  ): Promise<CollectionResponse<ProfitLossEvent>> => {
+    const response = await api.instance.post<ActionResult<CollectionResponse<ProfitLossEvent>>>(
+      `/reports/${payload.reportId}/data`,
+      snakeCaseTransformer(payload),
```

Comment:
Maybe just for correctness sake we can omit the `reportId` from the post body?

It's not needed there right since it's part of the path.